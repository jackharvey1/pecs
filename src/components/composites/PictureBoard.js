import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import CaptionedImage from './CaptionedImage';

const FlexRow = styled.div`
    display: inline-flex;
    flex-direction: row;
`;

const PictureBoard = ({ data }) => (
    <FlexRow>
        {data && data.map(({ src, alt, text }, i) => (
            <CaptionedImage
                key={i}
                src={src}
                alt={alt}
                text={text}
            />
        ))}
    </FlexRow>
);

PictureBoard.propTypes = {
    data: PropTypes.arrayOf(
        PropTypes.shape({
            src: PropTypes.string,
            alt: PropTypes.string,
            text: PropTypes.string
        })
    )
};

export default PictureBoard;

