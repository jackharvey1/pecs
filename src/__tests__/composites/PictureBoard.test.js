import React from 'react';
import { shallow } from 'enzyme';
import PictureBoard from '../../components/composites/PictureBoard';

describe('PictureBoard component', () => {
    it('renders as intended', () => {
        const wrapper = shallow(<PictureBoard
            data={[{
                src: 'http://image.com/image.png',
                text: 'Caption'
            }]}
        />);
        expect(wrapper).toMatchSnapshot();
    });
});
