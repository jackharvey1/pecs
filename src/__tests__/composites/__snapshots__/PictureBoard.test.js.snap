// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PictureBoard component renders as intended 1`] = `
<styled.div>
  <CaptionedImage
    key="0"
    src="http://image.com/image.png"
    text="Caption"
  />
</styled.div>
`;
